modules/taboolaBidAdapter.js

```diff
@@ -0,0 +1,255 @@
+'use strict';
+
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER} from '../src/mediaTypes.js';
+import {config} from '../src/config.js';
+import {getWindowSelf, getWindowTop} from '../src/utils.js'
+import {getStorageManager} from '../src/storageManager.js';
+
+const BIDDER_CODE = 'taboola';
+const GVLID = 42;
+const CURRENCY = 'USD';
+export const END_POINT_URL = 'https://taboolahb.bidder.taboolasyndication.com';
+const USER_ID = 'user-id';
+const STORAGE_KEY = `taboola global:${USER_ID}`;
+const COOKIE_KEY = 'trc_cookie_storage';
+
+/**
+ * try to extract User Id by that order:
+ *  local storage
+ *  first party cookie
+ *  rendered trc
+ *  new user set it to 0
+ */
+export const userData = {
+  storageManager: getStorageManager(GVLID, BIDDER_CODE),
+  getUserId: () => {
+    const {getFromLocalStorage, getFromCookie, getFromTRC} = userData;
+
+    try {
+      return getFromLocalStorage() || getFromCookie() || getFromTRC();
+    } catch (ex) {
+      return 0;
```

Comment:
what does this 0 stand for? isn't there anything we can do with the error?

Reply:
if we had an issue with getting the user id (you know browser stuff...) than we will go to fallback of user id = 0 that meens that the user is unknown